godal.cpp

```diff
@@ -1692,3 +1693,33 @@ void test_godal_error_handling(cctx *ctx) {
 	godalUnwrap();
 }
 
+void godalGridCreate(cctx *ctx, GDALGridAlgorithm eAlgorithm, void *poOptions, GUInt32 nPoints, const double *padfX, const double *padfY, const double *padfZ, double dfXMin,
+					double dfXMax, double dfYMin, double dfYMax, GUInt32 nXSize, GUInt32 nYSize, GDALDataType eType, void *pData) {
+	godalWrap(ctx);
+
+	CPLErr ret = GDALGridCreate(eAlgorithm, poOptions, nPoints, padfX, padfY,
+                      padfZ, dfXMin, dfXMax, dfYMin, dfYMax, nXSize, nYSize, 
+					  eType,  pData, nullptr, nullptr);
+	if(ret!=0) {
+		forceCPLError(ctx, ret);
+	}
+
+	godalUnwrap();
+}
+
+void godalGridParseAlgorithmAndOptions(cctx *ctx, char *pszAlgorithm, GDALGridAlgorithm *peAlgorithm, void **ppOptions) {
```

Comment:
There's no use in using 2 cgo calls here as there is no codepath that can call  godalGridParseAlgorithmAndOptions without before having called godalGridCreate. i.e. merge these 2 functions into a single one (cgo calls are "expensive" so we want to minimize the number of them)